components/director/internal/selfregmanager/selfreg_manager.go

```diff
@@ -85,12 +86,26 @@ func (s *selfRegisterManager) PrepareForSelfRegistration(ctx context.Context, re
 	}
 
 	if _, err := tenant.LoadFromContext(ctx); err == nil && consumerInfo.Flow.IsCertFlow() {
-		distinguishLabel, exists := labels[s.cfg.SelfRegisterDistinguishLabelKey]
-		if !exists {
+		if resourceType == resource.ApplicationTemplate {
+			labels[scenarioassignment.SubaccountIDKey] = consumerInfo.ConsumerID
+		}
+
+		distinguishLabel, distinguishLabelExists := labels[s.cfg.SelfRegisterDistinguishLabelKey]
+		_, productLabelExists := labels[s.appTemplateProductLabel]
+
+		switch {
+		case !distinguishLabelExists && !productLabelExists:
 			if resourceType == resource.Runtime {
 				return labels, nil
 			}
-			return nil, errors.Errorf("missing %q label", s.cfg.SelfRegisterDistinguishLabelKey)
+			return nil, errors.Errorf("missing %q or %q label", s.cfg.SelfRegisterDistinguishLabelKey, s.appTemplateProductLabel)
+		case distinguishLabelExists && productLabelExists:
+			if resourceType == resource.Runtime {
+				return labels, nil
+			}
+			return nil, errors.Errorf("should provide either %q or %q label - providing both at the same time is not allowed", s.cfg.SelfRegisterDistinguishLabelKey, s.appTemplateProductLabel)
+		case productLabelExists:
+			return labels, nil
```

Comment:
Do we want to return the current labels or to continue the execution of the function?

Reply:
If we continue, we will proceed with the self reg flow. As far as I understand, this is undesirable.